frontend/plan/src/components/search/SearchBar.js

```diff
@@ -37,6 +37,90 @@ function shouldSearch(filterData) {
     return searchString || selectedReq;
 }
 
+const LoginButton = () => (
+    <a
+        className="button is-link login"
+        href={"/accounts/login/?next=" + window.location}
+        style={{
+            padding: "0.5rem",
+            fontSize: "1rem!important",
+            paddingRight: "1rem",
+            paddingLeft: "1rem",
+        }}
+    >
+        Login
+    </a>
+);
+
+const UserSelector = ({ user: { first_name, last_name, username } }) => {
+    const [selected, setSelected] = useState(false);
+    const [ref, setRef] = useState(null);
+    useEffect(() => {
+        const listener = (event) => {
+            if (ref && !ref.contains(event.target)) {
+                setSelected(false);
+            }
+        };
+        document.addEventListener("click", listener);
+        return () => {
+            document.removeEventListener("click", listener);
+        };
+    });
+    return (
+        <div
+            className={`dropdown${selected ? " is-active" : ""}`}
+            ref={setRef}
+        >
+            <div
+                className={`dropdown-trigger${selected ? " user-selector-selected" : ""}`}
+                role="button"
+                id="user-selector"
+                onClick={() => setSelected(!selected)}
+            >
+                <span>
+                    {" "}
+                    {(first_name && first_name.charAt(0)) || "U"}
+                    {" "}
+                </span>
+            </div>
+            <div className="logout dropdown-menu">
+                <div id="logout-dropdown-menu-container">
+                    <div className="triangle-up"/>
+                    <div id="logout-dropdown-inner-menu">
+                        <p className="name-container">
+                            {" "}
+                            {first_name}
+                            {" "}
+                            {last_name}
+                            {" "}
+                        </p>
+                        <p className="email-container">
+                            {" "}
+                            {username}
+                            {" "}
+                        </p>
+                        <div
+                            role="button"
+                            id="logout-button"
+                            onClick={() => {
+                                fetch("/accounts/logout").then(console.log);
```

Comment:
From my debugger, seems like backend is responding with a 302. Code as is does not redirect - I think we need to add a redirect: "follow".
[https://stackoverflow.com/questions/39735496/redirect-after-a-fetch-post-call](url)

Reply:
can also fix by adding a trailing slash I'm 90% sure

Reply:
I tried `redirect: "follow"`, the trailing slash, and some other things I found on StackOverflow, but none of them worked. For now, I'll just explicitly check for a 302 and handle that.

Reply:
Actually, it looks like that's not even necessary. It still gets picked up in the `.then` block since it's not an error code, and it ends up performing the logout correctly anyway.